x-pack/plugins/enterprise_search/public/applications/app_search/components/api_logs/api_logs_logic.ts

```diff
@@ -0,0 +1,156 @@
+    dataLoading: [
+      true,
+      {
+        updateView: () => false,
+        onPaginate: () => true,
+      },
+    ],
+    apiLogs: [
+      [],
+      {
+        updateView: (_, { results }) => results,
+      },
+    ],
+    meta: [
+      DEFAULT_META,
+      {
+        updateView: (_, { meta }) => meta,
+        onPaginate: (state, { newPageIndex }) => updateMetaPageIndex(state, newPageIndex),
+      },
+    ],
+    hasNewData: [
+      false,
+      {
+        storePolledData: () => true,
+        updateView: () => false,
+      },
+    ],
+    polledData: [
+      {} as ApiLogsData,
+      {
+        storePolledData: (_, data) => data,
+      },
+    ],
+    intervalId: [
+      null,
+      {
+        onPollStart: (_, { intervalId }) => intervalId,
+      },
+    ],
+  }),
+  listeners: ({ actions, values }) => ({
+    pollForApiLogs: () => {
+      if (values.intervalId) return; // Ensure we only have one poll at a time
+
+      const id = window.setInterval(() => actions.fetchApiLogs({ isPoll: true }), POLLING_DURATION);
+      actions.onPollStart(id);
+    },
+    fetchApiLogs: async ({ isPoll }) => {
+      if (isPoll && values.dataLoading) return; // Manual fetches (i.e. user pagination) should override polling
+
+      const { http } = HttpLogic.values;
+      const { engineName } = EngineLogic.values;
+
+      try {
+        const response = await http.get(`/api/app_search/engines/${engineName}/api_logs`, {
+          query: {
+            'page[current]': values.meta.page.current,
+            'filters[date][from]': getDateString(-1),
+            'filters[date][to]': getDateString(),
+            sort_direction: 'desc',
+          },
+        });
+
+        // Manual fetches (e.g. page load, user pagination) should update the view immediately,
+        // while polls are stored in-state until the user manually triggers the 'Refresh' action
+        if (isPoll) {
+          actions.onPollInterval(response);
+        } else {
+          actions.updateView(response);
+        }
+      } catch (e) {
+        if (isPoll) {
+          // If polling fails, it will typically be due due to http connection -
+          // we should send a more human-readable message if so
+          setErrorMessage(POLLING_ERROR_MESSAGE);
+        } else {
+          flashAPIErrors(e);
+        }
+      }
+    },
+    onPollInterval: (data, breakpoint) => {
+      breakpoint(); // Prevents errors if logic unmounts while fetching
+
+      const previousResults = values.meta.page.total_results;
+      const newResults = data.meta.page.total_results;
+      const isEmpty = previousResults === 0;
+      const hasNewData = previousResults !== newResults;
+
+      if (isEmpty && hasNewData) {
+        actions.updateView(data); // Empty logs should automatically update with new data without a manual action
+      } else if (hasNewData) {
+        actions.storePolledData(data); // Otherwise, store any new data until the user manually refreshes the table
```

Comment:
I'm not sure if all these comments are helpful vs. just checking the unit tests - definitely let me know if you have thoughts there. The UX here is significantly different from anything else in our app so I think it could make sense to have that extra context in both places.